### PR TITLE
Ports clean_coords and its test

### PR DESCRIPTION
**Pull request by armantorkzaban**

### CODE REVIEW

File: lib/src/clean_coords.dart [hunk 1]
```diff
@@ -0,0 +1,355 @@
+// To-Do => Improve Typescript GeoJSON handling
+
+import '../helpers.dart';
+import 'invariant.dart';
+
+/// Removes redundant coordinates from any GeoJSON Geometry.
+/// Takes geojson Feature or Geometry
+/// [options.mutate=false] allows GeoJSON input to be mutated
```
Comment (lukas-h) [inline lib/src/clean_coords.dart:8]: it is just `mutate`

**Review by lukas-h — CHANGES_REQUESTED** [review]

File: lib/src/clean_coords.dart [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+      (geojson.geometry as GeometryType).coordinates = newCoords;
+      return geojson;
+    }
+    return Feature(
+      geometry: (geom as GeometryType)..coordinates = newCoords,
+      properties: geojson.properties,
+      bbox: geojson.bbox,
+      id: geojson.id,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (armantorkzaban) [reply lib/src/clean_coords.dart:78]: Does't line 54 suffice?

File: lib/src/clean_coords.dart [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+// To-Do => Improve Typescript GeoJSON handling
+
+/**
+ * Removes redundant coordinates from any GeoJSON Geometry.
+ *
+ * @name cleanCoords
+ * @param {Geometry|Feature} geojson Feature or Geometry
+ * @param {Object} [options={}] Optional parameters
+ * @param {boolean} [options.mutate=false] allows GeoJSON input to be mutated
```
[the remark this answers is not in the document]
Comment (armantorkzaban) [reply lib/src/clean_coords.dart:168]: Will do when I can test the function. Meanwhile keeping it as a reference.

File: lib/src/clean_coords.dart [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    list.forEach(
+      (element) {
+        if (!set.contains([element.alt, element.lat, element.lng].join('-'))) {
+          newCoords.add(element.clone());
+        }
+        set.add([element.alt, element.lat, element.lng].join('-'));
+      },
+    );
+  } else {
+    throw Exception("$geom is not supported");
```
Comment (lukas-h) [inline lib/src/clean_coords.dart:54]: Please use something like `${geom.type}`

File: lib/src/clean_coords.dart (hunk 1 above)
Comment (lukas-h) [reply lib/src/clean_coords.dart]: docs not changed yet.

File: lib/src/clean_coords.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukas-h) [reply lib/src/clean_coords.dart:78]: Yes you are right

File: lib/src/clean_coords.dart [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+
+  if (isPointOnLineSegment(newPoints[newPointsLength - 3],
+      newPoints[newPointsLength - 1], newPoints[newPointsLength - 2])) {
+    newPoints.removeAt(newPoints.length - 2);
+  }
+  return newPoints;
+}
+
+/// Compares two points and returns if they are equals
+equals(Position pt1, Position pt2) {
```
[the remark this answers is not in the document]
Comment (lukas-h) [reply lib/src/clean_coords.dart]: please don't resolve a conversation, if it is not done. Is there a reason we need this function?

**Review by lukas-h — CHANGES_REQUESTED** [review]

**Comment by lukas-h** [comment]
you can continue here, just merged #111

**Review by lukas-h — APPROVED** [review]

**Pull request merged**